### PR TITLE
fix: Test the playwright tests in Github actions

### PR DESCRIPTION
**Pull request by ivasilov**

### CODE REVIEW

**Comment by jcmorrow** [comment]
Hey @ivasilov! I've been working on recording the playwright test on this repo with [Replay](https://www.replay.io/). I was planning on putting up a PR soon, would that be a helpful tool for you trying to fix these? If so I'd be happy to pair on it and show you what I have so far!

File: apps/studio/pages/api/projects/[ref]/run-lints.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { NextApiRequest, NextApiResponse } from 'next'
```
Comment (saltcod) [inline apps/studio/pages/api/projects/[ref]/run-lints.ts:1]: this file a mistake in this branch?

**Review by saltcod — APPROVED** [review]

File: packages/ui/src/components/NavMenu/index.tsx [hunk 2]
```diff
@@ -20,17 +20,16 @@ export const NavMenu = forwardRef<HTMLDivElement, NavMenuProps>(
...
+  extends PropsWithChildren<{
     className?: string
     active: boolean
-  }>) => (
+  }> {}
+
+export const NavMenuItem = forwardRef<HTMLLIElement, NavMenuItemProps>(
+  ({ children, className, active, ...props }, ref) => (
     <li
+      ref={ref}
```
Comment (joshenlim) [inline packages/ui/src/components/NavMenu/index.tsx:32]: is this part of the playwright stuffs too?

File: playwright-tests/tests/snapshot/spec/table-editor.spec.ts [hunk 3]
```diff
@@ -1,30 +1,155 @@
...
+
+const dismissToast = async (page: Page) => {
+  await page.locator('#toast').getByRole('button').waitFor({ state: 'visible' })
+  await page.locator('#toast').getByRole('button').click()
+}
 
 test.describe('Table Editor page', () => {
-  test('should create a column and insert a row', async ({ page }) => {
-    const name = 'TestTable' + Math.floor(Math.random() * 10)
+  test.setTimeout(60000)
```
Comment (joshenlim) [inline playwright-tests/tests/snapshot/spec/table-editor.spec.ts:10]: any reason why we need a 60s timeout before the test runs?

File: packages/ui/src/components/NavMenu/index.tsx (hunk 2 above)
Comment (ivasilov) [reply packages/ui/src/components/NavMenu/index.tsx:32]: It logged an console error, so I fixed it to remove it. Could go in a separate PR also.

File: apps/studio/pages/api/projects/[ref]/run-lints.ts (hunk 1 above)
Comment (ivasilov) [reply apps/studio/pages/api/projects/[ref]/run-lints.ts:1]: When running in local environment,  the endpoint returns 404 so I added a API page route to silence the 404.

File: playwright-tests/tests/snapshot/spec/table-editor.spec.ts [hunk 4]
```diff
@@ -1,30 +1,155 @@
...
+    await page
+      .locator('.col-span-8 > div > .relative > .peer\\/input')
+      .first()
+      .waitFor({ state: 'visible' })
     await page.locator('.col-span-8 > div > .relative > .peer\\/input').first().click()
-    await page.locator('.col-span-8 > div > .relative > .peer\\/input').first().fill(name)
+    await page.locator('.col-span-8 > div > .relative > .peer\\/input').first().fill(tableName)
+
+    // make the built-in created_at column nullable
+    await page.locator('div:nth-child(3) > div > div > div:nth-child(7) > button').click()
```
[the remark this answers is not in the document]
Comment (ivasilov) [reply playwright-tests/tests/snapshot/spec/table-editor.spec.ts:36]: I agree, `data-test-id` is the more sustainable solution. I'll fix these.

[the remark this answers is not in the document]
Comment (saltcod) [reply playwright-tests/tests/snapshot/spec/table-editor.spec.ts:36]: Curious what's the guidance takeaway here? 
when should you use a data-test-id vs when can you use a class/role?

**Comment by ivasilov** [comment]
> Hey @ivasilov! I've been working on recording the playwright test on this repo with [Replay](https://www.replay.io/). I was planning on putting up a PR soon, would that be a helpful tool for you trying to fix these? If so I'd be happy to pair on it and show you what I have so far!

Hey @jcmorrow! I've fixed the flaky tests but it took me awhile to find a good way to stabilize them. I would love to see what you have. Send me a mail at ivan@supabase.io.

**Pull request merged**